packages/react/hooks.ts

```diff
@@ -232,11 +232,14 @@ export const useDeferredValue = (value: any) => {
 
 // useSyncExternalStore
 export const useSyncExternalStore = (subscribe, getSnapshot) => {
-  const state = useState(getSnapshot());
+  const [, updateState] = useState();
+  const forceUpdate = useCallback(() => updateState({}), []);
+  const [subscriber, setSubscriber] = useState(() => subscribe(forceUpdate));
+
   useEffect(() => {
-    subscribe(state);
-  });
-  subscribe(state);
+    subscriber?.();
+    setSubscriber(subscribe(forceUpdate));
+  }, [getSnapshot()]);
   return state;
```

Comment:
Where does the `state` come from if `const state = useState(getSnapshot());` is removed?

Reply:
I think we can go ahead with copying Preact's implementation.